src/app/lib/business/forced_update/update_required_service.dart

```diff
@@ -0,0 +1,37 @@
+import 'dart:async';
+
+import 'package:app/access/forced_update/current_version_repository.dart';
+import 'package:app/access/forced_update/minimum_version_repository.dart';
+
+/// This service checks if the application is running the minimum required version.
+abstract interface class UpdateRequiredService {
+  factory UpdateRequiredService(
+    MinimumVersionRepository minimumVersionRepository,
+    CurrentVersionRepository currentVersionService,
+  ) = _UpdateRequiredService;
+
+  /// Checks if the application is running the minimum required version.
+  /// We use a Future<void> since this is a one-time check.
+  Future<void> checkUpdateRequired();
```

Comment:
Shouldn't this be an event or a stream?

Reply:
Events are not really like in c#, and we could use a stream, but I thought that a one-off event would be represented well by a Future, since well, it does exactly what we want, in that we can wait until what we want to happen happens, and when it does, just do the thing we want.

Reply:
For a one-off thing, it seemed more appropriate to me than a stream.

Reply:
Oh I see. In that case, I wonder if we should change the name to something like `WaitForUpdateRequired`?

Reply:
Because the thing is, it may never happen.